js/ding_place2book.js

```diff
@@ -3,11 +3,23 @@
  *
  * Provide event nodes/pages with ticket info from Place2book
  */
-jQuery(document).ready(function($) {
-	$('.place2book-ticketinfo').each(function() { 
-	    var obj = this;
-		$.getJSON(Drupal.settings.basePath + 'ding/place2book/ticketinfo/' + this.value, function(data) {
-		  $(obj).replaceWith(data['markup']);
-		});		
-    });
-});
\ No newline at end of file
+(function ($) {
+
+  Drupal.behaviors.ding_place2book = {
+    attach: function (context, settings) {
+      $('.place2book-ticketinfo').once('place2book-button', function() {
+        var request = $.ajax({
+          url: Drupal.settings.basePath + 'ding/place2book/ticketinfo/' + this.value,
+          type: 'POST',
+          dataType: 'json',
+          success: ding_place2book_insert,
+        });
+      });
+    }
+  };
+
+  var ding_place2book_insert = function(ding_place2book) {
+    $('.place2book-ticketinfo').replaceWith(ding_place2book.markup);
```

Comment:
I can not be sure that it is as the modules is used on different site with different designs, so just to be sure and it would be simple to send the context as one more parameter to ding_place2book_insert.

One more thing that i just spotted ding_place2book_insert is missing an function documentation block?